File: src/main/node/net/ecma/Library/Tests/Components/Navigation/ViewpointJump.mjs
'use strict';
var X3D = require('./x3d.mjs');
try {
	var fs = require('fs');
} catch (e) {
console.log("Problems loading fs. On browser?",e);
}
var SFString = require('./x3d.mjs');
var SFNode = require('./x3d.mjs');
var head = require('./x3d.mjs');
var MFNode = require('./x3d.mjs');
var meta = require('./x3d.mjs');
var Scene = require('./x3d.mjs');
var Viewpoint = require('./x3d.mjs');
var SFVec3f = require('./x3d.mjs');
var TimeSensor = require('./x3d.mjs');
var SFTime = require('./x3d.mjs');
var SFBool = require('./x3d.mjs');
var Transform = require('./x3d.mjs');
var Shape = require('./x3d.mjs');
var Appearance = require('./x3d.mjs');
var Material = require('./x3d.mjs');
var Sphere = require('./x3d.mjs');
var Script = require('./x3d.mjs');
var field = require('./x3d.mjs');
var Text = require('./x3d.mjs');
var MFString = require('./x3d.mjs');
var NavigationInfo = require('./x3d.mjs');
var ROUTE = require('./x3d.mjs');
var X3D0 =  new X3D({

      profile : new SFString("Immersive"),
      version : new SFString("3.3"),
      head : new SFNode(
        new head({
          meta : new MFNode([
            new meta({
              name : new SFString("comment"),
              content : new SFString("World of Titania")}),

            new meta({
              name : new SFString("created"),
              content : new SFString("Wed, 05 Jul 2017 08:02:41 GMT")}),

            new meta({
              name : new SFString("creator"),
              content : new SFString("Holger Seelig")}),

            new meta({
              name : new SFString("generator"),
              content : new SFString("Titania V3.0.4, http://titania.create3000.de")}),

            new meta({
              name : new SFString("identifier"),
              content : new SFString("file:///home/holger/Projekte/Cobweb/excite/tests/Components/Navigation/ViewpointJump.x3d")}),

            new meta({
              name : new SFString("modified"),
              content : new SFString("Wed, 05 Jul 2017 08:10:01 GMT")})])})),
      Scene : new SFNode(
        new Scene({
          children : new MFNode([
            new Viewpoint({
              position : new SFVec3f([0,0,3.1])}),

            new TimeSensor({
              DEF : new SFString("TimeSensor1"),
              cycleInterval : new SFTime(3),
              loop : new SFBool(true)}),

            new Transform({
              children : new MFNode([
                new Shape({
                  appearance : new SFNode(
                    new Appearance({
                      material : new SFNode(
                        new Material({}))})),
                  geometry : new SFNode(
                    new Sphere({}))})])}),

            new Viewpoint({
              DEF : new SFString("Viewpoint1"),
              position : new SFVec3f([0,0,10.2])}),

            new Script({
              DEF : new SFString("Script1"),
              field : new MFNode([
                new field({
                  type : field.TYPE_SFFLOAT,
                  name : new SFString("float1_in"),
                  accessType : new SFString(field.ACCESSTYPE_INPUTONLY)}),

                new field({
                  type : field.TYPE_SFBOOL,
                  name : new SFString("bool1_out"),
                  accessType : new SFString(field.ACCESSTYPE_OUTPUTONLY)}),

                new field({
                  type : field.TYPE_MFSTRING,
                  name : new SFString("mstring1_out"),
                  accessType : new SFString(field.ACCESSTYPE_OUTPUTONLY)}),
              ]),javascript:

 , function float1_in(value)
    {
    // value  SFFloat
    if (value > 0.90)
        bool1_out = true;
    mstring1_out = new MFString(value);
    })}),

            new Transform({
              translation : new SFVec3f([-2.57105,0,0]),
              children : new MFNode([
                new Shape({
                  appearance : new SFNode(
                    new Appearance({
                      material : new SFNode(
                        new Material({}))})),
                  geometry : new SFNode(
                    new Text({
                      DEF : new SFString("Text1"),
                      string : new MFString(["0.6444690227508545"])}))})])}),

            new NavigationInfo({}),

            new ROUTE({
              fromNode : new SFString("TimeSensor1"),
              fromField : new SFString("fraction_changed"),
              toNode : new SFString("Script1"),
              toField : new SFString("float1_in")}),

            new ROUTE({
              fromNode : new SFString("Script1"),
              fromField : new SFString("bool1_out"),
              toNode : new SFString("Viewpoint1"),
              toField : new SFString("set_bind")}),

            new ROUTE({
              fromNode : new SFString("Script1"),
              fromField : new SFString("mstring1_out"),
              toNode : new SFString("Text1"),
              toField : new SFString("set_string")})])}))});
console.log(X3D0.toXMLNode());

File: src/main/c/net/coderextreme/Library/Tests/Components/Navigation/ViewpointJump.c
#include "C:/x3d-code/www.web3d.org/x3d/languages/c/Concretes.h"
void main(int argc, char ** argv) {
Browser browser = X3D.getBrowser();
X3D X3D0;
X3D0.profile = "Immersive";
X3D0.version = "3.3";
head head1 = createNode("head");
meta meta2 = createNode("meta");
meta2.name = "comment";
meta2.content = "World of Titania";
head1.meta = new MFNode();

head1.meta[0] = meta2;

meta meta3 = createNode("meta");
meta3.name = "created";
meta3.content = "Wed, 05 Jul 2017 08:02:41 GMT";
head1.meta[1] = meta3;

meta meta4 = createNode("meta");
meta4.name = "creator";
meta4.content = "Holger Seelig";
head1.meta[2] = meta4;

meta meta5 = createNode("meta");
meta5.name = "generator";
meta5.content = "Titania V3.0.4, http://titania.create3000.de";
head1.meta[3] = meta5;

meta meta6 = createNode("meta");
meta6.name = "identifier";
meta6.content = "file:///home/holger/Projekte/Cobweb/excite/tests/Components/Navigation/ViewpointJump.x3d";
head1.meta[4] = meta6;

meta meta7 = createNode("meta");
meta7.name = "modified";
meta7.content = "Wed, 05 Jul 2017 08:10:01 GMT";
head1.meta[5] = meta7;

head = head1;

Viewpoint Viewpoint9 = createNode("Viewpoint");
Viewpoint9.position = new SFVec3f(new float[0,0,3.1]);
children = new MFNode();

children[0] = Viewpoint9;

TimeSensor TimeSensor10 = createNode("TimeSensor");
TimeSensor10.DEF = "TimeSensor1";
TimeSensor10.cycleInterval = 3;
TimeSensor10.loop = True;
children[1] = TimeSensor10;

Transform Transform11 = createNode("Transform");
Shape Shape12 = createNode("Shape");
Appearance Appearance13 = createNode("Appearance");
Material Material14 = createNode("Material");
Appearance13.material = Material14;

Shape12.appearance = Appearance13;

Sphere Sphere15 = createNode("Sphere");
Shape12.geometry = Sphere15;

Transform11.child = new undefined();

Transform11.child[0] = Shape12;

children[2] = Transform11;

Viewpoint Viewpoint16 = createNode("Viewpoint");
Viewpoint16.DEF = "Viewpoint1";
Viewpoint16.position = new SFVec3f(new float[0,0,10.2]);
children[3] = Viewpoint16;

Script Script17 = createNode("Script");
Script17.DEF = "Script1";
field field18 = createNode("field");
field18.name = "float1_in";
field18.accessType = "inputOnly";
field18.type = "SFFloat";
Script17.field = new MFNode();

Script17.field[0] = field18;

field field19 = createNode("field");
field19.name = "bool1_out";
field19.accessType = "outputOnly";
field19.type = "SFBool";
Script17.field[1] = field19;

field field20 = createNode("field");
field20.name = "mstring1_out";
field20.accessType = "outputOnly";
field20.type = "MFString";
Script17.field[2] = field20;


Script17.setSourceCode(`javascript:\n"+
"\n"+
" function float1_in(value)\n"+
"    {\n"+
"    // value  SFFloat\n"+
"    if (value > 0.90)\n"+
"        bool1_out = true;\n"+
"    mstring1_out = new MFString(value);\n"+
"    }`)
children[4] = Script17;

Transform Transform21 = createNode("Transform");
Transform21.translation = new SFVec3f(new float[-2.57105,0,0]);
Shape Shape22 = createNode("Shape");
Appearance Appearance23 = createNode("Appearance");
Material Material24 = createNode("Material");
Appearance23.material = Material24;

Shape22.appearance = Appearance23;

Text Text25 = createNode("Text");
Text25.DEF = "Text1";
Text25.string = new MFString(new java.lang.String["0.6444690227508545"]);
Shape22.geometry = Text25;

Transform21.child = new undefined();

Transform21.child[0] = Shape22;

children[5] = Transform21;

NavigationInfo NavigationInfo26 = createNode("NavigationInfo");
children[6] = NavigationInfo26;

ROUTE ROUTE27 = createNode("ROUTE");
ROUTE27.fromNode = "TimeSensor1";
ROUTE27.fromField = "fraction_changed";
ROUTE27.toNode = "Script1";
ROUTE27.toField = "float1_in";
children[7] = ROUTE27;

ROUTE ROUTE28 = createNode("ROUTE");
ROUTE28.fromNode = "Script1";
ROUTE28.fromField = "bool1_out";
ROUTE28.toNode = "Viewpoint1";
ROUTE28.toField = "set_bind";
children[8] = ROUTE28;

ROUTE ROUTE29 = createNode("ROUTE");
ROUTE29.fromNode = "Script1";
ROUTE29.fromField = "mstring1_out";
ROUTE29.toNode = "Text1";
ROUTE29.toField = "set_string";
children[9] = ROUTE29;

}
